core/tests/protractor_utils/ExplorationEditorPage.js

```diff
@@ -0,0 +1,798 @@
+    by.css('.protractor-test-confirm-delete-response'));
+  var confirmDeleteStateButton = element(
+    by.css('.protractor-test-confirm-delete-state'));
+  var confirmDiscardChangesButton = element(
+    by.css('.protractor-test-confirm-discard-changes'));
+  var deleteAnswerButton = element(
+    by.css('.protractor-test-delete-answer'));
+  var deleteInteractionButton = element(
+    by.css('.protractor-test-delete-interaction'));
+  var deleteResponseButton = element(
+    by.css('.protractor-test-delete-response'));
+  var discardChangesButton = element(
+    by.css('.protractor-test-discard-changes'));
+  var dismissWelcomeModalButton = element(
+    by.css('.protractor-test-dismiss-welcome-modal'));
+  var navigateToFeedbackTabButton = element(
+    by.css('.protractor-test-feedback-tab'));
+  var navigateToHistoryTabButton = element(
+    by.css('.protractor-test-history-tab'));
+  var navigateToMainTabButton = element(by.css('.protractor-test-main-tab'));
+  var navigateToPreviewTabButton = element(
+    by.css('.protractor-test-preview-tab'));
+  var navigateToSettingsTabButton = element(
+    by.css('.protractor-test-settings-tab'));
+  var saveAnswerButton = element(
+    by.css('.protractor-test-save-answer'));
+  var saveChangesButton = element(by.css('.protractor-test-save-changes'));
+  var saveDiscardToggleButton = element(
+    by.css('.protractor-test-save-discard-toggle'));
+  var saveDraftButton = element(by.css('.protractor-test-close-save-modal'));
+  var saveInteractionButton = element(
+    by.css('.protractor-test-save-interaction'));
+  var saveOutcomeDestButton = element(
+    by.css('.protractor-test-save-outcome-dest'));
+  var saveOutcomeFeedbackButton = element(
+    by.css('.protractor-test-save-outcome-feedback'));
+  var saveStateContentButton = element(
+    by.css('.protractor-test-save-state-content'));
+  var stateNameSubmitButton = stateNameContainer.element(
+    by.css('.protractor-test-state-name-submit'));
+
+  /*
+   * Actions
+   */
+  // This clicks the "add new response" button and then selects the rule type
+  // and enters its parameters, and closes the rule editor. Any number of rule
+  // parameters may be specified after the ruleName.
+  // - interactionId: the name of the interaction type, e.g. NumericInput.
+  // - feedbackInstructions: a rich-text object containing feedback, or null.
+  // - destStateName: the name of the destination state of the rule, or null if
+  //     the rule loops to the current state.
+  // - createState: true if the rule creates a new state, else false.
+  // - refresherExplorationId: the id of refresher exploration for the current
+  //     state (if applicable), by default, should be null.
+  // - ruleName: the name of the rule, e.g. IsGreaterThan.
+  //
+  // Note that feedbackInstructions may be null (which means 'specify no
+  // feedback'), and only represents a single feedback element.
+  this.addResponse = function(
```

Comment:
Shouldn't this be in a "main tab" component? Ditto for some of the other stuff in this file.

Reply:
I have thought about this. Shouldn't the "main tab" be the ExplorationEditor page object? In particular, this addResponse method populates details on the "Add Response" template after creating a new interaction that requires response(s). 

If a new `ExplorationEditorMainTab` is created, `ExplorationEditorPage` would only contain 6 methods(`getMainTab`,`getHistoryTab`,...) with the `ExplorationEditorMainTab` object having 95% of the codes from `ExplorationEditorPage` object. Is that what you have in mind?

Reply:
Yes, I think ExplorationEditorMainTab should be different from the ExplorationEditorPage object -- that is more semantically correct. The latter is indeed likely to be smaller, but it also gives us a place to put methods that span multiple tabs (in addition to the ones you mention).

Reply:
Got it! In [workflow.js](https://github.com/oppia/oppia/blob/43dd04b54f47868931b79b95d9f7504d8f1b6f94/core/tests/protractor_utils/workflow.js#L59) and [general.js](https://github.com/oppia/oppia/blob/43dd04b54f47868931b79b95d9f7504d8f1b6f94/core/tests/protractor_utils/general.js#L116) , there are  a few methods that span multiple editor's tab. Would you want those moved to the new `ExplorationEditorPage`?

Reply:
Yes, but only if they don't interact with any other pages. Thanks!